src/transformers/pipelines/pt_utils.py

```diff
@@ -140,7 +143,7 @@ def __next__(self):
                 # elements.
                 self.loader_batch_size = observed_batch_size
             # Setting internal index to unwrap the batch
-            self._loader_batch_data = processed
+            self._loader_batch_data = processed[0] if isinstance(processed, tuple) else processed
```

Comment:
that I am less conviced, it's too general of a change

Reply:
it's for self.loader_batch_item() which only accept tensor or dict like input.